lib/streamlit/delta_generator.py

```diff
@@ -355,16 +361,37 @@ def wrapper(*args: Any, **kwargs: Any) -> NoReturn:
         return wrapper
 
     @property
-    def _parent_block_types(self) -> Iterable[str]:
+    def _parent_block_types(self) -> ParentBlockTypes:
         """Iterate all the block types used by this DeltaGenerator and all
         its ancestor DeltaGenerators.
         """
         current_dg: Optional[DeltaGenerator] = self
         while current_dg is not None:
             if current_dg._block_type is not None:
-                yield current_dg._block_type
+                yield current_dg._block_type, current_dg._root_container
             current_dg = current_dg._parent
 
+    def _get_raw_parent_block_types(
+        self, parent_block_types: ParentBlockTypes
+    ) -> FrozenSet[str]:
+        parent_block_types_raw = []
+        for parent_block, _ in parent_block_types:
+            parent_block_types_raw.append(parent_block)
+        return frozenset(parent_block_types_raw)
+
+    def _count_num_of_parent_columns(self, parent_block_types: ParentBlockTypes) -> int:
+        num_of_columns = 0
+        for parent_block, _ in parent_block_types:
+            if "column" == parent_block:
+                num_of_columns = num_of_columns + 1
+        return num_of_columns
```

Comment:
nit: This could be simplified to

```python
return sum(
    1 for parent_block, _ in parent_block_types if parent_block == "column"
)
```

at which point I wouldn't even mind not having this be a helper function and instead just in-lining the expression in the `_block` method below

Reply:
I've decided to left the func, since it's name says a lot about the behaviour, I've used you code snippet
now it's:
```
return sum(
    1 for parent_block in parent_block_types if parent_block == "column"
)
```
Thank you 🙇